### ISSUE TITLE
Accessibility Testing for Design System & Forms Library - Form Upload

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/va.gov-team#120631]
[Starting URL for testing](https://staging.va.gov/find-forms/upload/21-0779/introduction) 

### Product information

- [x] Team name, product name, and feature name have been added to the title of this issue.
- [x] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@jeana-adhoc 

### Use of color and color contrast checks

- [X] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [X] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [X] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [X] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [X] Any charts, maps, infographics, and tables convey all information without only relying on color
- [X] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")
- [ ] (Mobile app only) Color contrast ratios are met for all supported dark, contrast, and inverted color modes
- [ ] (Mobile app only) All elements retain their meaning for all supported dark, contrast, and inverted color modes

### How did color testing go?

Only known issue:
-  https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4683
  - This work is planned in a later epic:
    - https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3750)
### axe checks

- [X] Each page or screen has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

| Axe Results   | Page           |
| ------------- | -------------- |
| No issues  | [Introduction](https://staging.va.gov/find-forms/upload/21-0779/introduction) |
| No issues  | [Veteran's name and postal code](https://staging.va.gov/find-forms/upload/21-0779/name-and-zip-code) |  | No issues  | [Veteran's identification](https://staging.va.gov/find-forms/upload/21-0779/identification-information)  |
| No issues  | [Phone & Email](https://staging.va.gov/find-forms/upload/21-0779/phone-number-and-email) |
| No issues  | [File upload](https://staging.va.gov/find-forms/upload/21-0779/upload) |
| No issues  | [Supporting docs](https://staging.va.gov/find-forms/upload/21-0779/supporting-documents) |
| No issues  | [Upload supporting documentation](https://staging.va.gov/find-forms/upload/21-0779/upload-supporting-documents) |
| No issues | [Review page](https://staging.va.gov/find-forms/upload/21-0779/review-and-submit)

### How did axe testing go?

Axe identified no issues

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [X] (Web only) All page elements are readable and usable at 200% zoom
- [X] (Web only) All page elements are readable and usable at 300% zoom
- [X] (Web only) All page elements are readable and usable at 400% zoom
- [ ] (Mobile app only) All page elements are readable and usable when device text is at the maximum possible size

### How did content zoom and reflow testing go?

No issues

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [ ] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

We continue to have issues with focus displaying on errors 
 - https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4993

#### Form Issues

1. `/introduction` Should the Authenticated intro page have the [prefill alert](https://design.va.gov/components/form/prefill) on it? 

2. `/identification-information` In the on the Veteran identification wrap each sentence in a `<p>` tag so there is space between the 2 paragraphs instead of just a single `<br />` tag. - [#5186](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5186)

<img width="563" height="246" alt="Image" src="https://github.com/user-attachments/assets/6fbd2931-a112-4438-8388-9f3ef9eff1fb" />

3. `/identification-information` Remove the hint text on va file number, this is duplicate hint text due to the hint text in the `<legend>`- [#5184](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5184)

4. `supporting-documents` Error message needs updating. Change to something like "Select yes to upload your supporting documents" - [#5185](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5185)
<img width="563" height="349" alt="Image" src="https://github.com/user-attachments/assets/8c2b7c70-702e-4ec1-b182-4a40aa016f3b" />

5. `/upload` The Upload page is missing an `h3` every other page has an `h3`. I recognize there's an h3 in the alert - but it doesn't follow all of the other pages before it where there was an `h3` on the page and the alert was in the legend - [#5183](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5183)
    
    It seems we had a regression between this PR 
    - https://github.com/department-of-veterans-affairs/vets-website/commit/f1263deda14882a100d7f837c7dae13f5710e9f4
    
    And this PR
    - https://github.com/department-of-veterans-affairs/vets-website/pull/38753/files#diff-3489a623050bb30b7b21d78b91d3aac3680c59c1a344385cf9394ec4f5d06db6

6. `a11y-defect-2` - `/review-and-submit` Each `edit` link needs to have an accessible name. The first three have accessible names - [#5188](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5188)
    - Edit personal information
    - Edit contact information
    - Edit Uploaded file
    
     But the newly aded `Uploaded supporting documents``edit` link does not have an accessible name, so a screen reader just announces `Edit`. The link needs `label="Edit Uploaded supporting documents"` added to it.

7. `/review-and-submit` The `Edit` link needs relative font sizes on it. This is a known issue [FIND ISSUE LINK]. When browser sizes change the font size to "very large" the edit links do not resize as they should. - [#5187](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5187)

    <img width="400" alt="Image" src="https://github.com/user-attachments/assets/0dddcf37-49f5-4c36-b8d8-e55772449eec" />

8. `/review-and-submit` & `/supporting-documents`. When the answer to the question on the supporting document page is no, the `upload-supporting-documents` page is skipped. However, when you reach the review page, the `/supporting-documents` question/answer is not displayed, and you only see a file-input box on the review page.  I would expect to see "Do you want to upload supporting documents now? No" on the review page. No other fields allow you to edit/upload content directly to the review page in this form. All editing should take place back within the form itself. - [#5189](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5189)
    
    <img width="500" height="191" alt="Image" src="https://github.com/user-attachments/assets/89baf91e-949f-4151-8e9d-d777309cc3b3" />

9. **POTENTIALLY LAUNCH BLOCKING** `/upload`  -  When the page loads, there is a warning alert reminding users to sign their form before they upload it. When a user uploads their form, OCR Scanners do magic to determine if this looks like the form they intended to upload. If it does not look like the form they intendd, the page’s alert dynamically changes to show a new message, but this change isn’t announced to screen reader users. Because the new alert communicates a meaningful update (the result of an action) without being programmatically conveyed, it fails WCAG 2.2 – 4.1.3 (Status Messages). In short, critical feedback is visually visible but not perceivable to assistive technologies, leaving screen reader users unaware that the page has changed. - [#5191](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5191)

10. `/upload` - When I dismiss the initial alert that reminds users to sign their form, and then upload a file that does not look like 21-0779, the new alert notifying me that the file uploaded doesn't look like a recent VA Form 21-0799 loads *without* a heading. This is unexpected. - [#5192](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5192)

    <img width="575" height="740" alt="Image" src="https://github.com/user-attachments/assets/32a3506b-920a-4f88-9d46-a7c6d3b51d5a" />

11. `/name-and-zip-code` When dismissing the alert with the keyboard and tabbing to the first name field, I still hear the entire info alert announced again
    - NVDA - [#5193](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5193)

#### Component Issues
1. `va-link` on `/introduction` & `/upload` & `/confirmation`  There is a known bug in the download link. Right now, when you use the `download` prop, it adds the download icon, and downloads the file (it may be into the browser window or the download folder). OR you can have the download link open in a new tab using the `external` prop, and this will add text that says `(opens in a new tab)` You can't have it do both. I think for now, it's more important to not have users lose their place in the form, so we should use the `external` prop until the bug is fixed. 

2. `va-file-input-multiple` on `/upload-supporting-documents` The label configuration in the multiple file input is different than the single file input. Odd that the (* Required) Label is on its own line. Also when you enlarge browser font sizes, (* Required) does not resize in the multiple file upload - [#5194](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5194)

    **/component-library/packages/web-components/src/components/va-file-input/va-file-input.tsx**

    <img width="400" alt="Image" src="https://github.com/user-attachments/assets/ad6751f8-37e5-4ed0-9895-249743b16010" />

    ```html
    return (
            <div class="label-header">
              <label htmlFor="fileInputField" part="label" class="usa-label">
                {label}
                {requiredSpan}
              </label>
            </div>
          );
    ```

    **/component-library/packages/web-components/src/components/va-file-input-multiple/va-file-input-multiple.tsx**

    <img width="400"  alt="Image" src="https://github.com/user-attachments/assets/d929d287-7bce-44b5-b6ba-6f83fe542e58" />

    ```html
    return (
            <div class="label-header">
              <span part="label" class="usa-label">{label}</span>
              {requiredSpan}
            </div>
          );
    ```
3. **POTENTIALLY LAUNCH BLOCKING BUT KNOWN AND BEING WORKED ON** `va-file-input` & `va-file-input-multiple` on `upload` & `upload-supporting-documents` Error messages display, but are not announced and are not always focused as expected in the following scenarios
    - Error not announced when not uploading a file and clicking continue
      -  JAWS Announces label & hint text then error message; focus went to the entire frame of the compnent
    - Error not announced on file upload when error is triggered due to wrong filetype (link to error issues)
    - Error not announced when uploading a file that is too large
        - JAWS when dragging the file into the component it announces the error 
    - Error not announced when a duplicate file is uploaded in `va-file-input-multiple`
        - JAWS loses focuses and reads the page `<title>` and never announces the error 
    - Error not announced when uploading an empty file
      - JAWS Did announce error message first when dragging the file into the component 

4. `va-file-input` & `va-file-input-multiple` on `upload` & `upload-supporting-documents` When navigating through the page and through the file-input component, VO browse mode can interact with 'hidden' items  - [#5195](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5195)

    https://github.com/user-attachments/assets/b2d3b3ca-69bc-4efa-b268-1df172007b65

5. `va-file-input` & `va-file-input-multiple` on  `upload` & `upload-supporting-documents`  Some PDFs become scrollable when uploaded, but I cannot click into them, and I don't see to be able to get to them with the keyboard (VO) - Need to confirm with JAWS and NVDA -[ #5196](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5196)

https://github.com/user-attachments/assets/14f86674-43bb-4508-a89b-1e322962e615

6. `va-file-input` & `va-file-input-multiple` on `upload` & `upload-supporting-documents` When an empty file is uploaded, the error message says "files must be larger than 0B". This is confusing to hear, and should say "0 bytes".  Perhaps `<span aria-label="File size: 0 bytes">0 B</span>` - [#5197](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5197)

    <img width="540" height="368" alt="Image" src="https://github.com/user-attachments/assets/c96fad87-7503-465d-92c3-e49f92136976" />

7. `va-file-input` & `va-file-input-multiple` on `upload` & `upload-supporting-documents` Sometimes there is a "Forbidden" error message that appears. This error message needs more details as to what is wrong, and how the user can fix the problem.  I've generally encountered this in a way that makes me think this is a back-end glitch. Could the message simply say "There was a problem with our system. Delete this file and try again." - [#5198](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5198)
  
    <img width="564" height="405" alt="Image" src="https://github.com/user-attachments/assets/85bb9023-e2b5-42a4-aaed-2b355bcb3d11" />

8. `va-file-input` & `va-file-input-multiple` Progress indicator in the component continues announcing every 3% long after the UI has indicated the file has been uploaded - [#5199](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5199)
 
#### Forms Library

1. `review-and-submit` When getting an error submitting the form, I saw a [floating period in this file](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms/save-in-progress/SaveFormLink.jsx#L113). (it's not a speck of dust on your monitor). But, I also realized that the `Save your form` `link` isn't a link at all, it's really a button. We need to either architecturally make it a link, or make it look like a secondary button.

    <img width="600" alt="Image" src="https://github.com/user-attachments/assets/75d9a414-edb9-4d97-bdac-3552b585dbe8" />

### Known Existing Issues
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4993
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/120700

**Comment by jeana-adhoc** [comment]
This task is completed, the work called out in this issue is being handled in separate tickets.
